codersdk/users.go

```diff
@@ -176,6 +176,14 @@ type UpdateUserProfileRequest struct {
 	Name     string `json:"name" validate:"user_real_name"`
 }
 
+type ValidateUserPasswordRequest struct {
+	Password string `json:"password" validate:"required"`
+}
+
+type ValidateUserPasswordResponse struct {
+	Valid bool `json:"valid"`
```

Comment:
suggestion: as above, if we include the error it will be easier for the user to adjust their input.